### PR TITLE
feat: support sum(LinearMixin)

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This enables the use of `sum(...)` for subclasses of the `LinearMixin` class. It also adds the reflective operand dunder methods `__radd__` and `__rsub__`.

### Details and comments

Since this is not part of the publicly documented API, I did not add a reno.

### CODE REVIEW

File: qiskit/quantum_info/operators/mixins/linear.py
Comment (jakelishman) [inline qiskit/quantum_info/operators/mixins/linear.py:62]: Did you mean `(-self)._add`? It'd be good to add tests of the new from-the-right magic methods, but other than your scalar 0, I'm not sure they'd ever be successfully called?

**Review by jakelishman — COMMENTED** [review]
Mathematics aside, this looks good to me. It wouldn't hurt to have a feature release note saying that "`BaseOperator` subclasses (like x, y and z) can now be used with `sum`" - the magic methods aren't in the documentation, but the effects can still be seen by users.

File: qiskit/quantum_info/operators/mixins/linear.py
Comment (mrossinek) [reply qiskit/quantum_info/operators/mixins/linear.py:62]: The right-side magic methods would get called when adding two custom classes, where only one of them might support the addition with the other one. I could create a custom class for testing if you want. Or I test it using the magic `0` case again...

Comment (mrossinek) [reply qiskit/quantum_info/operators/mixins/linear.py]: I pushed some updates but it is still not ideal in terms of testing. Let me know what you think..

**Review by jakelishman — COMMENTED** [review]
Maybe to test _rsub_ we just call the method directly? We can compare `a.__sub__(b)` and `b.__rsub__(a)`. I think we can trust Python to delegate to do the magic delegation correctly, so manually calling seems cleanest. The reason to have more than `0 - a` is because that wouldn't have caught the original typo.

**Review by jakelishman — APPROVED** [review]
Great, thanks Max.  I pushed a commit to add the Sphinx crossrefs to the release note, and we're good to go.

**Comment by mrossinek** [comment]
Thanks, Jake, for the speedy review process! :pray:

**Comment by ikkoham** [comment]
LGTM (after merged haha)

**Comment by jakelishman** [comment]
Sorry - I thought it'd be inconsequential enough not to need to worry you about it!  (And I didn't expect you to be at work at 23:00...)

**Pull request merged**
